release/release.py

```diff
@@ -100,37 +101,18 @@ def failed_preconditions_changelog(bump: str, root: pathlib.Path) -> list[str]:
 
 def main() -> None:
     """Create the release."""
-    os.environ["RELEASE_DATE"] = datetime.date.today().isoformat()  # Used by bump2version to update CHANGELOG.md
-    current_version, bump, check_preconditions_only = parse_arguments()
+    os.environ["RELEASE_DATE"] = datetime.date.today().isoformat()  # Used by bump-my-version to update CHANGELOG.md
+    bump, check_preconditions_only = parse_arguments()
     check_preconditions(bump)
     if check_preconditions_only:
-        return
-    commands = []
+       return
     if bump.startswith("rc-"):
-        rc_bump, non_rc_bump = bump.split("-")
-        commands.append(["bump2version", "--no-tag", "--no-commit", non_rc_bump])
-        message = f"Bump version: {current_version} → {{new_version}}"
-        # Previous command makes the work space dirty, so allow dirty in the next version bump
-        commands.append(
-            [
-                "bump2version",
-                "--config-file",
-                ".bumpversion-rc.cfg",
-                "--message",
-                message,
-                "--tag-message",
-                message,
-                "--allow-dirty",
-                rc_bump,
-            ]
-        )
+        bump = bump.split("-")[1]  # Create a patch, minor, or major release candidate
     elif bump == "drop-rc":
-        new_version = current_version.split("-")[0]
-        commands.append(["bump2version", "--new-version", new_version, PART])
-    else:
-        commands.append(["bump2version", bump])
-    for command in commands:
-        subprocess.run(tuple(command), check=True)
+        bump = "pre_release_label"  # Bump the pre-release label from "rc" to "final" (which is optional and omitted)
```

Comment:
I don't understand the comment; in order to delete the release candidate, we make the rc final? Or does it mean that the process doesn't do anything when the value "final" is passed; if so, how is the rc cleaned up, and/or can't we just short-circuit `return` here?

Reply:
All releases without `rc` are `final`, but the final part of the version number is optional. Apparently this is how bump-my-version implements parts of the version number that are not always present.

Reply:
I'll add a link to the bump-my-version docs.